core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/ModeStateMachineServices.java

```diff
@@ -336,6 +396,30 @@ public EObject showHideMSMTransitions(DSemanticDecorator view, List<StateTransit
     return view;
   }
 
+  private boolean isIn(AbstractDNode node, DDiagram diag) {
+    return node.getParentDiagram() == diag;
+  }
+
+  public AbstractDNode getMappingMSMStateMode(ECrossReferenceAdapter croffRef, AbstractState state, DDiagram diagram) {
+    final String mappingName;
```

Comment:
looks to be DiagramServices.getDiagramElement(diagram, state).getActualMapping

Reply:
Yes I am rewritting this methods.